core/lib/types/src/storage/witness_block_state.rs

```diff
@@ -5,8 +5,43 @@ use serde::{Deserialize, Serialize};
 use crate::{StorageKey, StorageValue};
 
 /// Storage data used during Witness Generation.
-#[derive(Debug, Default, Serialize, Deserialize)]
+#[derive(Debug, Default, Clone)]
 pub struct WitnessBlockState {
     pub read_storage_key: HashMap<StorageKey, StorageValue>,
     pub is_write_initial: HashMap<StorageKey, bool>,
 }
+
+/// A serde schema for serializing/deserializing `WitnessBlockState`
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+struct WitnessBlockStateSerde {
+    pub read_storage_key: Vec<(StorageKey, StorageValue)>,
+    pub is_write_initial: Vec<(StorageKey, bool)>,
+}
```

Comment:
Hmm, I probably miss something, please explain why converting to Vec is needed for (de)serializing

Reply:
Yeah, I faced this issue in the runtime, serde isn't able to serialize HashMap's where the keys are some structs - read as it is not able to turn StorageKey to some string key(of course we can implement Display or something similar for it, but then it will be harder to deserialize)